src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -240,9 +241,19 @@ export default class Preview extends Setup {
 
         const component = this.flags.componentname;
 
-        const configFilePath = path.resolve(configFileName);
-        const configFile = PreviewUtils.loadConfigFile(configFilePath);
-        const appConfig = configFile.getAppConfig(platform, targetApp);
+        let appConfig:
+            | IOSAppPreviewConfig
+            | AndroidAppPreviewConfig
+            | undefined;
+
+        if (
```

Comment:
I think the addition of this block of code is fine. It falls into the realm of input validation, conditional data setup, etc. But I feel like the logic [below](https://github.com/forcedotcom/lwc-dev-mobile/pull/67/files#diff-63566dbef9589f4600f2ae97ce90eefdR270-R278) is starting to bleed into the platform-specific business logic of handling previews. That functionality should really be in the domain of the respective platform launchers (`IOSLauncher`, `AndroidLauncher`).

I think a straightforward way to move these decisions down the line—and what feels cleaner to me anyway—is that we should be passing the entire `appConfig` along to the respective launchers, rather than parsing it out here and sending individual config parameters along through the method calls. The platform-specific classes should be deciding how to manage their data inputs. For the most part they already are, but I think they should be handling the app configuration decisions as well.